Sources/Attribution/AttributionPoster.swift

```diff
@@ -128,15 +128,37 @@ final class AttributionPoster {
     @available(tvOS, unavailable)
     @available(watchOS, unavailable)
     func postAdServicesTokenIfNeeded() {
-        guard latestNetworkIdAndAdvertisingIdentifierSent(network: .adServices) == nil else {
-            return
+        guard let attributionToken = self.adServicesTokenToPostIfNeeded else { return }
+
+        self.post(adServicesToken: attributionToken)
+    }
+
+    var adServicesTokenToPostIfNeeded: String? {
+        #if os(tvOS) || os(watchOS)
+        return nil
+        #else
+        guard #available(iOS 14.3, macOS 11.1, macCatalyst 14.3, *) else {
+            return nil
         }
 
-        guard let attributionToken = attributionFetcher.adServicesToken else {
-            return
+        guard self.latestNetworkIdAndAdvertisingIdentifierSent(network: .adServices) == nil else {
```

Comment:
I've not been involved in the specs so this might be correct, but with this, only the first purchase for a user will send the token right? As in, if a user unsubscribes, the purchase expires and then resubscribes, the second time it won't be sent. Not sure if this token can change, so it might be ok to just send it the first time, but I thought I would mention it.

Reply:
yeah, that should be fine. That's the same behavior we have for adservices in other places. 
And in practice, attribution tokens expire quickly, and a user will only be attributed once